glib-macros/src/downgrade_struct.rs

```diff
@@ -0,0 +1,44 @@
+use crate::downgrade_fields::{derive_downgrade_fields, DowngradeStructParts};
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+use syn::{Generics, Ident};
+
+pub fn derive_downgrade_for_struct(
+    ident: Ident,
+    generics: Generics,
+    data_struct: syn::DataStruct,
+) -> TokenStream {
+    let weak_ref = format_ident!("{}WeakRef", ident);
```

Comment:
I'd call it `{}Weak` without the `Ref` as it isn't really a reference in the Rust sense (`&x`)